scripts/upgrade_version.py

```diff
@@ -0,0 +1,85 @@
+import click
+import os
+import subprocess
+import sys
+
+RAY_ABSOLUTE_PATH = ""
+
+NON_JAVA_FILES = set([
+    'ci/ray_ci/utils.py',
+    'python/ray/_version.py',
+    'src/ray/common/constants.h',
+])
+
+JAVA_FILES = set([
+    'java/api/pom_template.xml',
+    'java/performance_test/pom_template.xml',
+    'java/pom.xml',
+    'java/runtime/pom_template.xml',
+    'java/serve/pom_template.xml',
+    'java/test/pom_template.xml',
+])
+
+NON_JAVA_VERSION = "3.0.0.dev0"
+JAVA_VERSION = "2.0.0-SNAPSHOT"
+NEW_VERSION = "3.0.0.dev0"
+
+def update_current_version():
+    """
+    Scan for current Ray version and update the version global variables.
+    """
+    global NON_JAVA_VERSION
+    global JAVA_VERSION
+    output = subprocess.run(['python', 'python/ray/_version.py'], stdout=subprocess.PIPE, text=True)
+
+    if output.returncode != 0 or not output.stdout:
+        raise Exception('Failed to get current version')
+
+    version = output.stdout.split(' ')[0]
+    if version != NON_JAVA_VERSION:
+        NON_JAVA_VERSION = version
+        JAVA_VERSION = version
+
+def find_ray_absolute_path():
+    """
+    Find the absolute path of the ray directory.
+    """
+    current_directory = os.path.dirname(__file__)
+    # Loop to parent directory until ray directory is found
+    while current_directory != os.path.dirname(current_directory): # Stop at root
+        if os.path.basename(current_directory) == 'ray':
+            return current_directory
+        current_directory = os.path.dirname(current_directory)
```

Comment:
when running with `bazelisk run`, bazel sets a env var called `BUILD_WORKSPACE_DIRECTORY` that points to the root of the repo. please just read and use that.

Reply:
fixed!